src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicStream.cs

```diff
@@ -176,7 +178,7 @@ internal MsQuicStream(MsQuicConnection.State connectionState, QUIC_STREAM_OPEN_F
                 QuicExceptionHelpers.ThrowIfFailed(status, "Failed to open stream to peer.");
 
                 Debug.Assert(!Monitor.IsEntered(_state), "!Monitor.IsEntered(_state)");
-                status = MsQuicApi.Api.StreamStartDelegate(_state.Handle, QUIC_STREAM_START_FLAGS.FAIL_BLOCKED);
+                status = MsQuicApi.Api.StreamStartDelegate(_state.Handle, QUIC_STREAM_START_FLAGS.FAIL_BLOCKED | QUIC_STREAM_START_FLAGS.SHUTDOWN_ON_FAIL);
```

Comment:
Since the stream start is asynchronous now, do we need to do anything to prevent stream operations before we get the StartComplete callback?

Reply:
Most possibly, but I believe it will be part of https://github.com/dotnet/runtime/issues/67302 (I also mentioned it in additional thoughts there)

Reply:
> behavior depending on max stream limit is broken now

only throwing on reaching the limit is broken, right? the wait loop implemented in H3 should be fine? @ManickaP

Reply:
That's my understanding.

Reply:
FYI, it might be worth having a conversation about the `FAIL_BLOCKED` usage here. MsQuic allows you to start a stream and send on it before starting a connection (practically a requirement for 0-RTT scenarios), and without this flag it allows you to queue safely. But with this flag, it would just fail because you are blocked on starting the connection. It _might_ make sense to use this in your HTTP/3 scenarios, but not universally.

Reply:
Yes, we do plan to rework stream starting in #67302. We have more issues around this than just 0-RTT :smile: